libgadget/treewalk.c

```diff
@@ -342,10 +363,16 @@ static int ev_primary(TreeWalk * tw)
     }
 
     ev_alloc_threadlocals();
-#pragma omp parallel
+
+    int nint = tw->Ninteractions;
+    int nnodes = tw->Nnodesinlist;
+#pragma omp parallel reduction(+: nint) reduction(+: nnodes)
     {
-        real_ev(tw);
+        real_ev(tw, &nint, &nnodes);
```

Comment:
The reason real_ev is moved into a function was to help GDB identifying the local variables in a omp block.  I think newer versions of gdb doesn't have this issue. So free free to merge real_ev back here.

Reply:
Both real_ev and ev_primary are quite long functions. I'm happy to keep them separate.